allauth_2fa/adapter.py

```diff
@@ -22,9 +22,15 @@ def login(self, request, user):
             request.session['allauth_2fa_user_id'] = str(user.id)
 
             redirect_url = reverse('two-factor-authenticate')
-            # Add GET parameters to the URL if they exist.
-            if request.GET:
-                redirect_url += u'?' + urlencode(request.GET)
+            # Add "next" parameter to the URL.
+            view = request.resolver_match.func.view_class()
```

Comment:
Will this break for function-based views?

Reply:
@clokep I wanted to continue on this PR today and resolve this issue by writing a test that uses a function-based allauth login view, but the allauth login views are all class-based. Also, all of your other tests only use class-based views. Perhaps function-based views aren't really a supported concept of the allauth ecosystem?

Reply:
Additionally, in the release notes for allauth 0.13.0 (2013-08-31) it says
> All views are now class-based.

https://django-allauth.readthedocs.io/en/latest/release-notes.html#id112

Reply:
Huh, maybe it doesn't matter then!